### PR TITLE
Cache controller model instead of reallocating each detection

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview
* Adds a controller model cache for platform models, so we aren't reallocating and reloading the glTF model every time the same controller is detected.
* Removes the haptics extensions, as they referenced a DLL we don't currently distribute, so they don't currently work.
* Fixes some null propagation issues
* Removes some duplicate methods

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/5437

## Verification
Deploy an MRTK example scene app to your WMR machine. Open the app, and open a slate while inside the immersive experience. Move your controller's pointer between the slate and the immersive experience. You should see the controller flip back and forth between the shell's model and the app model without much delay.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs [hunk 1]
```diff
@@ -1,138 +1,49 @@
...
+                IReadOnlyList<SpatialInteractionSourceState> sources = null;
+
                 UnityEngine.WSA.Application.InvokeOnUIThread(() =>
                 {
-                    IReadOnlyList<SpatialInteractionSourceState> sources = SpatialInteractionManager.GetForCurrentView().GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+                    sources = SpatialInteractionManager.GetForCurrentView()?.GetDetectedSourcesAtTimestamp(PerceptionTimestampHelper.FromHistoricalTargetTime(DateTimeOffset.Now));
+                }, true);
 
-                    for (var i = 0; i < sources.Count; i++)
+                for (var i = 0; i < sources?.Count; i++)
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs]: Does i < null == false?

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs [hunk 2]
```diff
@@ -652,6 +652,12 @@ private void RemoveController(InteractionSource interactionSource)
             {
                 IMixedRealityInputSystem inputSystem = Service as IMixedRealityInputSystem;
 
+                if (controller.Visualizer != null &&
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs]: Not exactly sure what kind of tests one would add, but just popping question here...tests? (if applicable) :P

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs]: I thought about tests! But wasn't successful coming up with any. Most of these changes are only relevant at runtime on-device (the only place the `WindowsMixedRealityDeviceManager` will get a valid source detected), but I'll take a look and see if there's anything I can add.

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs (hunk 1 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs]: That's an interesting point! I'll double check that this functions expectedly

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Extensions/InteractionSourceExtensions.cs]: Under https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types#operators

>For the relational operators (<, >, <=, >=), if one or both operands are null, the result is false.

**Pull request merged**